### PR TITLE
First PR

### PR DESCRIPTION
**Pull request by Jzow**
This PR did about two things

1. Make constants for some fixed numbers, such as buffer_size
2. Remove un used crates components

I think we can separate the front and back ends so that the structure is relatively clear. I try to add GitHub Action CI to cargo, but I encounter some problems.

### CODE REVIEW

**Comment by Jzow** [comment]
By the way, we shouldn't leave `Cargo.lock` file

**Comment by kkoomen** [comment]
> By the way, we shouldn't leave `Cargo.lock` file

lock files should always be committed to ensure the version being run locally is the same on other environments. You should never put it in the gitignore.

File: src-tauri/src/file.rs [hunk 1]
```diff
@@ -1,15 +1,21 @@
 use std::io::{Write, Read};
 use std::fs::File;
 
+const BUFFER_SIZE: usize = 4096;
```
Comment (kkoomen) [inline src-tauri/src/file.rs:4]: None of this works. These variables are not available anymore in the scope of the `compress()` and `decompress()` functions. Rust compiler also complains about this.

File: src-tauri/src/file.rs [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+const Q: u32 = 11;
+const LGWIN: u32 = 22;
+
 pub fn compress(filename: &str, contents: &str) -> bool {
-    let mut writer = brotli::CompressorWriter::new(File::create(filename).unwrap(), 4096, 11, 22);
-    write!(&mut writer, "{}", contents).expect("Could not compress contents");
-    return true;
+    let writer = brotli::CompressorWriter::new(File::create(filename).unwrap(), BUFFER_SIZE, Q, LGWIN);
```
Comment (kkoomen) [inline src-tauri/src/file.rs:9]: must be mutable, you can't change this.

File: src-tauri/src/file.rs [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+    write!(writer, "{}", contents).expect("Could not compress contents");
+    
+    true
 }
 
 pub fn decompress(filename: &str) -> String {
-    let mut reader = brotli::Decompressor::new(File::open(filename).unwrap(), 4096);
+    let reader = brotli::Decompressor::new(File::open(filename).unwrap(), BUFFER_SIZE);
```
Comment (kkoomen) [inline src-tauri/src/file.rs:16]: must be mutable, you can't change this.

**Review by kkoomen — CHANGES_REQUESTED** [review]
I really doubt if you're making a PR to help out or just for the sake to be the first PR, because it seems like you didn't test at all. I get compiling errors for most of the things you've changed.

**Comment by Jzow** [comment]
@kkoomen If it is a tool provided for users, I think it should be retained. If we are developing lib, there should be no such file. Can you help withdraw the cargo lock file

**Comment by kkoomen** [comment]
> @kkoomen If it is a tool provided for users, I think it should be retained. If we are developing lib, there should be no such file. Can you help withdraw the cargo lock file

No, then you don't understand the meaning of a lock file. To ensure a stable version, people should install the versions of all the libs we've used. If people have no lock file then they might download a higher version which would maybe result in errors for the users. As I said, lock files should always be in the repository.

**Comment by Jzow** [comment]
@kkoomen I'm sorry, because of the lock file, I don't quite understand that users should always keep using the latest version

**Comment by kkoomen** [comment]
> @kkoomen I'm sorry, because of the lock file, I don't quite understand that users should always keep using the latest version

Even Tauri explicitly mentioned that it is required: https://tauri.studio/guides/development/development-cycle

I'm closing your issue as it doesn't add any value to this repo.

**Pull request closed without merging**